### PR TITLE
Develop ConvertLike Reference Implementation

### PR DESCRIPTION
**Pull request by ggalieroc**
Overview:

- Add reference implementation to evaluates_map() using reference::convert
- Enable single layer tests for GPU plugin
- Add backend tests

ticket: 37506

### CODE REVIEW

File: ngraph/core/include/ngraph/op/convert_like.hpp [hunk 1]
```diff
@@ -43,6 +43,9 @@ namespace ngraph
                 virtual std::shared_ptr<Node>
                     clone_with_new_inputs(const OutputVector& new_args) const override;
             };
-        }
-    }
-}
+
+        } // namespace v1
+        using v1::ConvertLike;
```
Comment (ilyachur) [inline ngraph/core/include/ngraph/op/convert_like.hpp:48]: Please remove this `using`. We have such `using`s only for some v0 operations and we don't want to support it in the future.

Comment (ggalieroc) [reply ngraph/core/include/ngraph/op/convert_like.hpp]: done

**Review by iefode — APPROVED** [review]

**Review by pelszkow — APPROVED** [review]
https://godbolt.org/z/Ej318K

**Comment by ggalieroc** [comment]
@pelszkow I added your suggestion into a new commit as it simplifies the code and improves readability

**Review by pelszkow — APPROVED** [review]

File: ngraph/test/runtime/interpreter/evaluates_map.cpp [hunk 2]
```diff
@@ -1102,91 +1092,109 @@ namespace
...
+        default: return false;
         }
-        else
+        return true;
+    }
+
+    namespace convert_like_v1
+    {
+        template <element::Type_t ti, element::Type_t to>
+        inline void evaluate(const shared_ptr<op::v1::ConvertLike>& op,
```
Comment (pelszkow) [inline ngraph/test/runtime/interpreter/evaluates_map.cpp:1141]: I not sure but in my opinion `enable_if` should be on this level. `element_type_traits<T>::value_type` for `T == boolean` return `char` which is one of `unsigned char` or `signed char` and for one of for `T == i8` or `T == u8` will have the same `value_type` as `boolean`. 
I'm not sure but for now it looks like values for `i8` or for `u8` will be reduce to `boolean` [0,1].
Am I right?

Comment (pelszkow) [reply ngraph/test/runtime/interpreter/evaluates_map.cpp:1141]: https://en.cppreference.com/w/cpp/language/types#Character_types

**Review by ilyachur — APPROVED** [review]

**Pull request merged**